### ISSUE TITLE
Optionally email me when my training job is done

### ISSUE THREAD

**Issue by matthewdzmura** [pytorch/torchtune#372]
As mentioned in https://github.com/pytorch-labs/torchtune/issues/371 I've been running my training jobs in the background. Sometimes there's an error and the job stops very early in the run, but I don't notice for a while until I check up on it and see that it's not running. Other times it completes but I don't realize it until I manually check which is itself tedious and slows down my iteration speed.

Something that would be really helpful would be to alert me via email when my job completes. If the job is successful this would include a summary of the training run e.g.
- Number of iterations
- Time spent running
- Loss curve
- Location of outputs like model checkpoint, logs, etc.
- Eval metrics or link to eval report if we have them

And if the job fails sending an email with the stack trace and a link to the logs would let me know I need to dig in and figure out what went wrong ASAP.
